packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -276,23 +278,29 @@ class ImageCache {
   /// [maximumSize] and [maximumSizeBytes].
   void _touch(Object key, _CachedImage image, TimelineTask? timelineTask) {
     assert(timelineTask != null);
-    if (image.sizeBytes != null && image.sizeBytes! <= maximumSizeBytes) {
+    if (image.sizeBytes != null && image.sizeBytes! <= maximumSizeBytes && maximumSize > 0) {
       _currentSizeBytes += image.sizeBytes!;
       _cache[key] = image;
       _checkCacheSize(timelineTask);
+    } else {
+      image.dispose();
     }
   }
 
-  void _trackLiveImage(Object key, _LiveImage image) {
+  void _trackLiveImage(Object key, ImageStreamCompleter completer) {
     // Avoid adding unnecessary callbacks to the completer.
     _liveImages.putIfAbsent(key, () {
       // Even if no callers to ImageProvider.resolve have listened to the stream,
       // the cache is listening to the stream and will remove itself once the
       // image completes to move it from pending to keepAlive.
       // Even if the cache size is 0, we still add this listener.
-      image.completer.addOnLastListenerRemovedCallback(image.handleRemove);
-      return image;
-    }).sizeBytes ??= image.sizeBytes;
+      return _LiveImage(
```

Comment:
The comment above here is confusing and/or outdated since no listener is added in the code below?

Reply:
Updated the comment.